x-pack/test/alerting_api_integration/spaces_only/tests/alerting/get_alert_state.ts

```diff
@@ -0,0 +1,89 @@
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import expect from '@kbn/expect';
+import { Spaces } from '../../scenarios';
+import { getUrlPrefix, ObjectRemover, getTestAlertData } from '../../../common/lib';
+import { FtrProviderContext } from '../../../common/ftr_provider_context';
+
+// eslint-disable-next-line import/no-default-export
+export default function createGetAlertStateTests({ getService }: FtrProviderContext) {
+  const supertest = getService('supertest');
+  const retry = getService('retry');
+
+  describe('getAlertState', () => {
+    const objectRemover = new ObjectRemover(supertest);
+
+    afterEach(() => objectRemover.removeAll());
+
+    it('should handle getAlertState request appropriately', async () => {
+      const { body: createdAlert } = await supertest
+        .post(`${getUrlPrefix(Spaces.space1.id)}/api/alert`)
+        .set('kbn-xsrf', 'foo')
+        .send(getTestAlertData())
+        .expect(200);
+      objectRemover.add(Spaces.space1.id, createdAlert.id, 'alert');
+
+      const response = await supertest.get(
+        `${getUrlPrefix(Spaces.space1.id)}/api/alert/${createdAlert.id}/state`
+      );
+
+      expect(response.statusCode).to.eql(200);
+      expect(response.body).to.key('alertInstances', 'previousStartedAt');
+    });
+
+    it('should fetch updated state', async () => {
+      const { body: createdAlert } = await supertest
+        .post(`${getUrlPrefix(Spaces.space1.id)}/api/alert`)
+        .set('kbn-xsrf', 'foo')
+        .send({
+          enabled: true,
+          name: 'abc',
+          tags: ['foo'],
+          alertTypeId: 'test.cumulative-firing',
+          consumer: 'bar',
+          schedule: { interval: '5s' },
+          throttle: '5s',
+          actions: [],
+          params: {},
+        })
+        .expect(200);
+      objectRemover.add(Spaces.space1.id, createdAlert.id, 'alert');
+
+      // wait for alert to actually execute
+      await retry.try(async () => {
+        const response = await supertest.get(
+          `${getUrlPrefix(Spaces.space1.id)}/api/alert/${createdAlert.id}/state`
+        );
+
+        expect(response.statusCode).to.eql(200);
+        expect(response.body).to.key('alertInstances', 'alertTypeState', 'previousStartedAt');
+        expect(response.body.alertTypeState.runCount).to.greaterThan(1);
+      });
+
+      const response = await supertest.get(
+        `${getUrlPrefix(Spaces.space1.id)}/api/alert/${createdAlert.id}/state`
+      );
+
+      expect(response.body.alertTypeState.runCount).to.greaterThan(0);
+
+      const alertInstances = Object.entries<Record<string, any>>(response.body.alertInstances);
+      expect(alertInstances.length).to.eql(response.body.alertTypeState.runCount);
+      alertInstances.forEach(([key, value], index) => {
+        expect(key).to.eql(`instance-${index}`);
+        expect(value.state).to.eql({ instanceStateValue: true });
```

Comment:
I wonder if the `instanceStateValue` should be something different, to be on the safe side. Perhaps the `index` value of the instance + `runCount`.